### PR TITLE
feat: authentication banner

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Made the `AuthOptions` accept a `className` object to be applied on certain elements.
- Introduced `ignoreMessages` prop. This is because we will now be rendering two `AuthOptions` component. One for the banner, and one for the modal. We want the banner to only be utilized when clicking the provider buttons. The other (modal) auth options will be listening to the messages which will continue the process.
- Tracking events are intact since we just reused the same component.

Preview in action:
https://dailydotdev.slack.com/archives/C06K1LYU716/p1707947399158029?thread_ts=1707940049.888179&cid=C06K1LYU716

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-169 #done

### CODE REVIEW

File: packages/shared/src/components/auth/AuthOptions.tsx
Comment (sshanzel) [inline packages/shared/src/components/auth/AuthOptions.tsx:359]: Fixed the `forceDefaultDisplay` to always show it, the primary reason we are forcing it, rather than having it in the initial state.

File: packages/shared/src/components/auth/AuthenticationBanner.tsx
Comment (nensidosari) [inline packages/shared/src/components/auth/AuthenticationBanner.tsx:15]: We might want to use similar setup for other experiments, I already know one coming up soon which had the same idea, so can we have this as a styled component maybe? We can leave the background part out of the styled component though as that might change.

File: packages/shared/src/components/auth/AuthenticationBanner.tsx
Comment (nensidosari) [inline packages/shared/src/components/auth/AuthenticationBanner.tsx:18]: Can we change the color to pepper 90?
```suggestion
        background: `linear-gradient(270deg, rgba(206, 61, 243, 0.16) 0%, rgba(113, 71, 237, 0.16) 100%), var(--Colors-Background-default, ${colors.pepper['90']})`,
```

File: packages/shared/src/components/auth/AuthenticationBanner.tsx
Comment (nensidosari) [inline packages/shared/src/components/auth/AuthenticationBanner.tsx:34]: We're adding a lot of !!! everywhere 😢

File: packages/shared/src/components/post/BasePostContent.tsx
Comment (nensidosari) [inline packages/shared/src/components/post/BasePostContent.tsx:34]: Glad another of thesee is removed

`    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
    // eslint-disable-next-line react-hooks/rules-of-hooks`

File: packages/shared/src/lib/featureManagement.ts
Comment (nensidosari) [inline packages/shared/src/lib/featureManagement.ts:38]: @nensidosari for me to remember to setup the experiment

File: packages/webapp/pages/onboarding.tsx
Comment (nensidosari) [inline packages/webapp/pages/onboarding.tsx:381]: Maybe to avoid the ! in the banner we're adding, we can actually accept the specific classnames for the title from both places.

**Review by nensidosari — COMMENTED** [review]
Few comments, but nothing blocking

File: packages/shared/src/components/auth/AuthenticationBanner.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx:18]: Ah, good point. Though I think we should use the one that varies based on theme. Will update this quickly.

File: packages/shared/src/components/auth/AuthenticationBanner.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx:15]: Yeah, I agree, though I've actually done that in a separate PR as I felt the need there 😅 

So maybe we can resolve this one out?
https://github.com/dailydotdev/apps/pull/2636/commits/97bde42f56ee86f9062c71670012e17d3f8b2749

File: packages/shared/src/components/auth/AuthOptions.tsx
Comment (rebelchris) [inline packages/shared/src/components/auth/AuthOptions.tsx:252]: Because of the ignore here don't you miss below set error?

File: packages/webapp/pages/onboarding.tsx
Comment (sshanzel) [reply packages/webapp/pages/onboarding.tsx:381]: You mean like adding a `typoClassName`?

File: packages/shared/src/components/auth/AuthenticationBanner.tsx
Comment (rebelchris) [inline packages/shared/src/components/auth/AuthenticationBanner.tsx:17]: Custom colors?
Maybe double verify with Tsahi, probably a design mistake?

**Review by rebelchris — CHANGES_REQUESTED** [review]
@sshanzel How does it now popup the modal on click?
Feels somehow I'm missing where this is invoked (*although see it working)

File: packages/shared/src/components/auth/AuthOptions.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthOptions.tsx:252]: Yes, we won't since the modal is opened, it has its own listener (with `ignoreMessages = false`) which allows the whole process to be completed.

File: packages/shared/src/components/auth/AuthenticationBanner.tsx
Comment (nensidosari) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx:15]: Ah yes, in the mobile PR, yes that resolves it. I would suggest though a more generic name like BottomBannerContainer. Do we use something similar for cookies banner as well?

**Comment by sshanzel** [comment]
> @sshanzel How does it now popup the modal on click?
Feels somehow I'm missing where this is invoked (*although see it working)

Through here:
https://github.com/dailydotdev/apps/blob/9c843023f384eaa98a5f8c43ebba80e572603484/packages/shared/src/components/auth/AuthenticationBanner.tsx#L35-L37

File: packages/webapp/pages/onboarding.tsx
Comment (nensidosari) [reply packages/webapp/pages/onboarding.tsx:381]: No I mean not having default typo setup in the classnames in OnboardingHeadline, but actually accepting the typos from classname props here just like we do on the other place we use OnboardingHeadline
`ex. className={{ title: '!typo-mega3', description: '!typo-title3' }}` If we do the same here we avoid the !

Comment (sshanzel) [reply packages/webapp/pages/onboarding.tsx:381]: Ah, yep. That makes sense. Let's do that 🚀

File: packages/shared/src/components/auth/AuthenticationBanner.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx:15]: Cool. Would it be okay for me to tackle that in the other PR?

File: packages/shared/src/components/auth/AuthOptions.tsx
Comment (rebelchris) [reply packages/shared/src/components/auth/AuthOptions.tsx:252]: Got it :)

File: packages/shared/src/components/auth/AuthenticationBanner.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx:17]: Sent a Slack message. Will wait for their response 🚢

File: packages/shared/src/components/auth/AuthenticationBanner.tsx
Comment (nensidosari) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx:15]: Yup

**Review by nensidosari — APPROVED** [review]
All good on my side 🚀

File: packages/shared/src/lib/featureManagement.ts
Comment (nensidosari) [reply packages/shared/src/lib/featureManagement.ts:38]: experiment feature is setup as well

Comment (sshanzel) [reply packages/shared/src/lib/featureManagement.ts:38]: Thank you!

File: packages/shared/src/components/auth/AuthenticationBanner.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx:17]: Found some equivalent values in our codebase, will update quickly.

Comment (sshanzel) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx:17]: Applied the missing styles 🥳 

Screenshot 2024-02-15 at 5 59 26 PM

Comment (rebelchris) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx:17]: Not urgent, but would it be possible to use the tailwind variation of this?
If too much work never mind :)

**Review by rebelchris — APPROVED** [review]
Non blocking comment, otherwise looks good to me :)

File: packages/shared/src/components/auth/AuthenticationBanner.tsx
Comment (sshanzel) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx:17]: Let's see, I will check tailwinds docs for linear gradient as it can definitely be used again in the future.

Comment (sshanzel) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx]: Doesn't look exactly the same after converting. Not sure if we go for this.

Screenshot 2024-02-15 at 6 35 35 PM

Comment (sshanzel) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx]: Merging now to unblock, but so far this is what I have tried:

```
 <BottomBannerContainer
      // style={style}
      className="gap-6 bg-gradient-to-l from-theme-overlay-active-cabbage from-0% via-theme-overlay-active-onion via-100% bg-theme-bg-primary py-10 shadow-3"
    >
```

I've also tried adding `to-theme-bg-primary` but doesn't change anything since the stop mentions 0 to 100 already. Not sure how it behaves differently on CSS style.

Comment (sshanzel) [reply packages/shared/src/components/auth/AuthenticationBanner.tsx]: Found the issue and fixed here: https://github.com/dailydotdev/apps/pull/2638

**Pull request merged**
